src/Tizen.Multimedia.Vision/MediaVision/DeepLearningFaceRecognizer.cs

```diff
@@ -0,0 +1,177 @@
+        /// <param name="label">The name of face data.</param>
+        /// <exception cref="ObjectDisposedException">The DeepLearningFaceRecognizer already has been disposed.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="source"/> or <paramref name="label"/> is null.</exception>
+        /// <exception cref="InvalidOperationException">In case of any invalid operations.</exception>
+        /// <since_tizen> 10 </since_tizen>
+        public void RegisterFace(MediaVisionSource source, string label)
+        {
+            ValidateNotDisposed();
+
+            if (source == null)
+            {
+                throw new ArgumentNullException(nameof(source));
+            }
+            if (label == null)
+            {
+                throw new ArgumentNullException(nameof(label));
+            }
+
+            InteropFace.Register(_handle, source.Handle, label).Validate("Failed to register face data");
+        }
+
+        /// <summary>
+        /// Unregisters face data from internal database.
+        /// </summary>
+        /// <param name="label">The name of face data.</param>
+        /// <exception cref="ObjectDisposedException">The DeepLearningFaceRecognizer already has been disposed.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="label"/> is null.</exception>
+        /// <exception cref="InvalidOperationException">In case of any invalid operations.</exception>
+        /// <since_tizen> 10 </since_tizen>
+        public void UnregisterFace(string label)
+        {
+            ValidateNotDisposed();
+
+            if (label == null)
+            {
+                throw new ArgumentNullException(nameof(label));
+            }
+
+            InteropFace.Unregister(_handle, label).Validate("Failed to register face data");
+        }
+
+        /// <summary>
+        /// Recognizes a face in by finding the closest match among the registered faces and returns the label of the found face.
+        /// </summary>
+        /// <param name="source">The face data to recognize.</param>
+        /// <returns>A label of recognized face.</returns>
+        /// <exception cref="ObjectDisposedException">The DeepLearningFaceRecognizer already has been disposed.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="source"/> is null.</exception>
+        /// <exception cref="InvalidOperationException">In case of any invalid operations.</exception>
+        /// <since_tizen> 10 </since_tizen>
+        public string Recognize(MediaVisionSource source)
+        {
+            ValidateNotDisposed();
+
+            if (source == null)
+            {
+                throw new ArgumentNullException(nameof(source));
+            }
+
+            InteropFace.Inference(_handle, source.Handle).Validate("Failed to recognize face");
+
+            InteropFace.GetLabel(_handle, out string label).Validate("Failed to get label");
```

Comment:
Could you consider separate API for GetLabel()?
In case of native CAPI, mv_face_recognition_get_label() is provided. By doing this, I think we could keep API consistency between C# and native API. In the future, other API for getting various types of results may be added.

You can refer to below test code,
https://review.tizen.org/gerrit/gitweb?p=platform/core/api/mediavision.git;a=blob;f=test/testsuites/machine_learning/face_recognition/test_face_recognition.cpp;h=f1be72d33bd76c344678a5f687fb8c76c922cc02;hb=refs/heads/tizen_devel#l191

Reply:
I added new class `DeepLearningFaceRecognitionResult` including recognized result, as we discussed in online.
If some result data is added more In the future, it could be added in this class as property easily.